### PR TITLE
Site Transfers: Add minor improvements to site transfer screen

### PR DESCRIPTION
**Pull request by katinthehatsite**
Related to https://github.com/Automattic/dotcom-forge/issues/2809

## Proposed Changes

This PR:

* Removes the list items for "Upgrades" since it is always only one item on the Transfer screen
* Change the second sentence in the "Content and ownership" section to: “You’ll keep your admin access unless {oldOwnerEmail} removes you.”
* Changes the third sentence to `Your posts on <strong>%(siteSlug)s</strong> will remain authored your account.`
* Change the texts to display the `username` in all cases except the first occurrence that shows `username (email)`.

**Before**:

Screenshot 2023-06-22 at 3 50 26 PM

**After**:

Screenshot 2023-06-22 at 3 50 52 PM

## Testing Instructions

* Navigate to **Settings > General > Transfer your site** with Calypso Live link
* Select an administrator to transfer your site to (if you don't have an admin on your site, you would need to add it first)
* Proceed to the transfer consent screen and confirm it looks as indicated by design

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx [hunk 1]
```diff
@@ -202,9 +200,9 @@ const ContentAndOwnershipCard = ( {
 							// translators: siteSlug is the current site slug, siteOwner is the user that the site is going to
 							// transer to
 							translate(
-								'You will remain as an administrator on <strong>%(siteSlug)s</strong>, unless removed by <strong>%(siteOwner)s</strong>.'
+								'You will keep your admin access unless <strong>%(siteOwner)s</strong> removes you.'
```
Comment (sejas) [inline client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx]: We could remove `siteSlug` from the translators comment.

File: client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx [hunk 2]
```diff
@@ -215,9 +213,9 @@ const ContentAndOwnershipCard = ( {
 							// translators: siteSlug is the current site slug, siteOwner is the user that the site is going to
 							// transer to
 							translate(
-								'Your posts on <strong>%(siteSlug)s</strong> will remain with your account, unless they are transferred to <strong>%(siteOwner)s</strong>.'
+								'Your posts on <strong>%(siteSlug)s</strong> will remain authored your account.'
```
Comment (sejas) [inline client/my-sites/site-settings/site-owner-transfer/start-site-owner-transfer.tsx]: We could remove `siteOwner` from the translators comment.

**Review by sejas — APPROVED** [review]
@katinthehatsite , I tested this PR works great.
I added a comment about editing the translators comment if you add/remove variables from there.
But no action is required since we will merge the other PR #78536 to this one.

**Comment by sejas** [comment]
@katinthehatsite , I merged my branch into this one, and now we show the `username` in all the sentences except the first one.

**Comment by katinthehatsite** [comment]
Thanks @sejas - I noticed one small typo in the content text so I will fix it now and merge the PR after!!

**Pull request merged**